gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -0,0 +1,752 @@
+				tiledMapTileMapObject.setScaleX(scaleX * (objectWidth / textureRegion.getRegionWidth()));
+				tiledMapTileMapObject.setScaleY(scaleY * (objectHeight / textureRegion.getRegionHeight()));
+				tiledMapTileMapObject.setRotation(element.getFloat("rotation", 0));
+				object = tiledMapTileMapObject;
+			} else {
+				object = new RectangleMapObject(x, flipY ? y - height : y, width, height);
+			}
+		}
+		object.setName(element.getString("name", null));
+		String rotation = element.getString("rotation", null);
+		if (rotation != null) {
+			object.getProperties().put("rotation", Float.parseFloat(rotation));
+		}
+		String type = element.getString("type", null);
+		if (type != null) {
+			object.getProperties().put("type", type);
+		}
+		int id = element.getInt("id", 0);
+		if (id != 0) {
+			object.getProperties().put("id", id);
+		}
+		object.getProperties().put("x", x);
+
+		if (object instanceof TiledMapTileMapObject) {
+			object.getProperties().put("y", y);
+		} else {
+			object.getProperties().put("y", (flipY ? y - height : y));
+		}
+		object.getProperties().put("width", width);
+		object.getProperties().put("height", height);
+		object.setVisible(element.getBoolean("visible", true));
+		JsonValue properties = element.get("properties");
+		if (properties != null) {
+			loadProperties(object.getProperties(), properties);
+		}
+		idToObject.put(id, object);
+		objects.add(object);
+
+	}
+
+	private void loadProperties (final MapProperties properties, JsonValue element) {
+		if (element == null) return;
+
+		if (element.name() != null && element.name().equals("properties")) {
+			for (JsonValue property : element) {
+				final String name = property.getString("name", null);
+				String value = property.getString("value", null);
+				String type = property.getString("type", null);
+				if (value == null) {
+					value = property.asString();
+				}
+				if (type != null && type.equals("object")) {
+					// Wait until the end of [loadTiledMap] to fetch the object
+					try {
+						// Value should be the id of the object being pointed to
+						final int id = Integer.parseInt(value);
+						// Create [Runnable] to fetch object and add it to props
+						Runnable fetch = new Runnable() {
+							@Override
+							public void run () {
+								MapObject object = idToObject.get(id);
+								properties.put(name, object);
+							}
+						};
+						// [Runnable] should not run until the end of [loadTiledMap]
+						runOnEndOfLoadTiled.add(fetch);
+					} catch (Exception exception) {
+						throw new GdxRuntimeException(
+							"Error parsing property [\" + name + \"] of type \"object\" with value: [" + value + "]", exception);
+					}
+				} else {
+					Object castValue = castProperty(name, value, type);
+					properties.put(name, castValue);
+				}
+			}
+		}
+	}
+
+	private Object castProperty (String name, String value, String type) {
+		if (type.equals("string")) {
```

Comment:
Ok, quick question.
So the tmx and tmj formats differ in this regard.
When tiled generates the TMJ file it explicitly denotes String properties as type="string" while in the TMX file there is no type added and that denotes a it as string, so in the castProperty() method for the TMX loader they look for nulls 
to mean String property.

So 2 options. 
1. We check for null in the TMJ loader and give the user the same "unsupported Type Message"
OR
2. We can cover both cases by doing this `if (type == null || "string".equals(type))  ` and then this can be another method added to our new refactored utility or parent class.

Reply:
I vote for option 2